core/kernel/huk_subkey.c

```diff
@@ -15,18 +15,31 @@ static TEE_Result mac_usage(void *ctx, uint32_t usage)
 }
 
 #ifdef CFG_CORE_HUK_SUBKEY_COMPAT
+#ifdef CFG_CORE_HUK_SUBKEY_COMPAT_USE_OTP_DIE_ID
```

Comment:
Please avoid `ifdef` where possible, it is preferred to use the `IS_ENABLED()` macro, like:
``` C
static TEE_Result get_otp_die_id(uint8_t *buffer, size_t len)
{
	if (IS_ENABLED(CFG_CORE_HUK_SUBKEY_COMPAT_USE_OTP_DIE_ID)) {
		return tee_otp_get_die_id(buffer, len);
	} else {
		static const char pattern...
		return TEE_SUCCESS;
	}
}

Reply:
Done.